lib/models/WebsiteConfiguration.ts

```diff
@@ -0,0 +1,215 @@
+    * @param [params.redirect.replaceKeyPrefixWith] - string to replace
+    *   keyPrefixEquals specified in condition
+    * @param [params.redirect.replaceKeyWith] - string to replace key
+    * @param [params.redirect.httpRedirectCode] - http redirect code
+    * @param [params.condition] - specifies conditions for a redirect
+    * @param [params.condition.keyPrefixEquals] - key prefix that
+    *   triggers a redirect
+    * @param [params.condition.httpErrorCodeReturnedEquals] - http code
+    *   that triggers a redirect
+    */
+    constructor(params?: RoutingRuleParams) {
+        if (params) {
+            this._redirect = params.redirect;
+            this._condition = params.condition;
+        }
+    }
+
+    /**
+    * Return copy of rule as plain object
+    * @return rule;
+    */
+    getRuleObject() {
+        const rule = {
+            redirect: this._redirect,
+            condition: this._condition,
+        };
+        return rule;
+    }
+
+    /**
+    * Return the condition object
+    * @return condition;
+    */
+    getCondition() {
+        return this._condition;
+    }
+
+    /**
+    * Return the redirect object
+    * @return redirect;
+    */
+    getRedirect() {
+        return this._redirect;
+    }
+}
+
+export type RedirectAllRequestsTo = {
+    hostName: string;
+    protocol?: string;
+};
+export class WebsiteConfiguration {
+    _indexDocument?: string;
+    _errorDocument?: string;
+    _redirectAllRequestsTo?: RedirectAllRequestsTo;
+    _routingRules?: RoutingRule[];
+
+    /**
+    * Object that represents website configuration
+    * @constructor
+    * @param params - object containing params to construct Object
+    * @param params.indexDocument - key for index document object
+    *   required when redirectAllRequestsTo is undefined
+    * @param [params.errorDocument] - key for error document object
+    * @param params.redirectAllRequestsTo - object containing info
+    *   about how to redirect all requests
+    * @param params.redirectAllRequestsTo.hostName - hostName to use
+    *   when redirecting all requests
+    * @param [params.redirectAllRequestsTo.protocol] - protocol to use
+    *   when redirecting all requests ('http' or 'https')
+    * @param params.routingRules - array of Routing
+    *   Rule instances or plain routing rule objects to cast as RoutingRule's
+    */
+    constructor(params: {
+        indexDocument: string;
+        errorDocument: string;
+        redirectAllRequestsTo: RedirectAllRequestsTo;
+        routingRules: RoutingRule[] | any[],
+    }) {
+        if (params) {
+            this._indexDocument = params.indexDocument;
+            this._errorDocument = params.errorDocument;
+            this._redirectAllRequestsTo = params.redirectAllRequestsTo;
+            this.setRoutingRules(params.routingRules);
+        }
+    }
+
+    /**
+    * Return plain object with configuration info
+    * @return - Object copy of class instance
+    */
+    getConfig() {
```

Comment:
Isn't there some typescript way to define return types in the code, like this? And if so should it be used?

```suggestion
    * @return - Object copy of class instance
    */
    getConfig(): Object {
```

Reply:
Sure we could, but as written, the type is directly infered by TypeScript. Try to build the project with TSC, you'll see the correct signature for the function, which is something like 
```typescript
type Return = { 
    indexDocument?: string;
    errorDocument?: string;
    redirectAllRequestsTo?: RedirectAllRequestsTo;
    routingRules?: {
        redirect: Redirect;
        condition: Condition;
    };
}
```